### PR TITLE
Events - Player events will report driver seat as turret [-1]

### PR DESCRIPTION
**Pull request by PabstMirror**
Replace #1312 
Reports driver/pilot as turret `[-1]`
This is mainly so it will also report pilot's weapon on "turretweapon" event

e.g. `["turretweapon", {x1 = +_this}, true] call CBA_fnc_addPlayerEventHandler;`
for AH-6 Littlebird will report `[player, "M134_minigun","missiles_DAR"]`

**This could cause problems for code that is not expecting driver's seat turret**
And many cars will report a weapon like `"TruckHorn2"` which might not be expected

### CODE REVIEW

File: addons/events/fnc_addPlayerEventHandler.sqf [hunk 1]
```diff
@@ -65,7 +65,7 @@ private _id = switch (_type) do {
     };
     case "turretweapon": {
         if (_applyRetroactively) then {
-            [GVAR(oldUnit), GVAR(oldUnit) currentWeaponTurret (GVAR(oldUnit) call CBA_fnc_turretPath), ""] call _function;
+            [GVAR(oldUnit), GVAR(oldTurretWeapon), ""] call _function;
```
Comment (commy2) [inline addons/events/fnc_addPlayerEventHandler.sqf]: Why?

File: addons/events/fnc_addPlayerEventHandler.sqf [hunk 2]
```diff
@@ -95,7 +95,7 @@ private _id = switch (_type) do {
     };
     case "turret": {
         if (_applyRetroactively) then {
-            [GVAR(oldUnit), GVAR(oldUnit) call CBA_fnc_turretPath, []] call _function;
+            [GVAR(oldUnit), GVAR(oldTurret), []] call _function;
```
Comment (commy2) [inline addons/events/fnc_addPlayerEventHandler.sqf]: Same. Why?

File: addons/events/fnc_addPlayerEventHandler.sqf (hunk 1 above)
Comment (PabstMirror) [reply addons/events/fnc_addPlayerEventHandler.sqf]: we have the value available in a variable, simplest and fastest way to return it

File: addons/events/fnc_addPlayerEventHandler.sqf (hunk 2 above)
Comment (PabstMirror) [reply addons/events/fnc_addPlayerEventHandler.sqf]: we're already using `GVAR(oldUnit)`, why not do the same for other vars

Comment (commy2) [reply addons/events/fnc_addPlayerEventHandler.sqf]: All other besides player calculate the current value.

Comment (PabstMirror) [reply addons/events/fnc_addPlayerEventHandler.sqf]: If all the other ones drove off a cliff, should this one as well?

**Pull request merged**